source/utilities.tex

```diff
@@ -16245,7 +16245,7 @@
 
 \pnum
 \throws
-Shall not throw exceptions when \tcode{f} is a function pointer
+Does not throw exceptions when \tcode{f} is a function pointer
```

Comment:
@jwakely wrote:

> The existing wording also applies to program-defined specializations of `duration` and `pair`. The proposed change makes that less clear IMO.

...And program-defined specializations of `function`, is that right? So essentially your view is that any library wording that describes the intended behavior of a function template or class template must always use "shall", not "does," because the user can always produce UB by mis-specializing a template? (A "mis-specialization" is a program-defined specialization that does anything different from what the library template is actually documented to do. For example, if the library template is documented to have a copy constructor, and the program-defined specialization does something other than copy; or if the library template is documented as "does not throw exceptions" and the program-defined specialization does throw exceptions.)

I thought SD-8 would have something to say about the inadvisability of mis-specializing library templates, but [the actual text of SD-8](https://isocpp.org/std/standing-documents/sd-8-standard-library-compatibility) seems ridiculously conservative on that score.

Reply:
> So essentially your view is that any library wording that describes the intended behavior of a function template or class template must always use "shall", not "does,"

No, I didn't say that. I'm just pointing out that this may not be as obvious as it appears. However, I think [namespace.std] p2 means that even if we remove these "shalls", program-defined specializations still have to meet that same requirements.

Reply:
So, @jwakely , is this good to go as presented? If so, please hit "approve".